src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/CourtEventChargeRepository.java

```diff
@@ -13,14 +13,26 @@ public interface CourtEventChargeRepository extends CrudRepository<CourtEventCha
         This also filters out data where there is missing court event data, caused by a
         NOMIS bug where orphan records were left when deleting court data.
      */
-    @Query("select " +
-        "cec " +
-        "from CourtEventCharge cec " +
-        "join cec.eventAndCharge.offenderCharge oc " +
-        "join oc.offenderBooking ob " +
-        "join ob.offender o " +
-        "where o.nomsId = :offenderId " +
-        "and exists (select id from CourtEvent where id = cec.eventAndCharge.courtEvent.id)"
+    @Query("""
+        select cec
+            from CourtEventCharge cec
+            join fetch cec.eventAndCharge eac
+            join fetch eac.courtEvent ce
+       left join fetch ce.courtEventType cet
+       left join fetch ce.eventStatus es
+       left join fetch ce.outcomeReasonCode
+            join fetch eac.offenderCharge oc
+            join fetch oc.offence off
+            join fetch oc.offenderCourtCase occ
+       left join fetch oc.resultCodeOne rco
+       left join fetch oc.offenderSentenceCharges osc
+       left join fetch osc.offenderSentence os
+       left join fetch os.courtOrder co
+            join       oc.offenderBooking ob
+            join       ob.offender o
+            where o.nomsId = :offenderId
+            and exists (select id from CourtEvent where id = ce.id)
+        """
```

Comment:
this doesnt actually eliminate multiple SQL calls entirely but it gets most of the worst offenders down to < 10